dev/tests/Unit/ComponentManagerTest.php

```diff
@@ -0,0 +1,182 @@
+<?php
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Dev\Tests\Unit;
+
+use Google\Cloud\Core\Testing\TestHelpers;
+use Google\Cloud\Dev\ComponentManager;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @group dev
+ */
+class ComponentManagerTest extends TestCase
+{
+    const ROOT_PATH = '/var/www';
+
+    private $cm;
+
+    public function setUp()
+    {
+        $this->cm = TestHelpers::stub(ComponentManager::class, [
+            self::ROOT_PATH
+        ], [
+            'components',
+            'manifest'
+        ]);
+
+        $this->cm->___setProperty('components', $this->components);
+        $this->cm->___setProperty('manifest', $this->manifest);
+    }
+
+    public function testComponents()
+    {
+        $this->assertEquals(array_keys($this->components), $this->cm->components());
+    }
+
+    public function testComponentsExtra()
+    {
+        $components = $this->components;
+        array_walk($components, function (&$component) {
```

Comment:
In other spots in the library, you use foreach loops. You could do the same thing and foreach by reference. I think consistency of functional vs imperative control structures would be nice.